sdk/nodejs/cmd/run/run.ts

```diff
@@ -53,14 +64,38 @@ function projectRootFromProgramPath(program: string): string {
     }
 }
 
+async function npmPackageRootFromProgramPath(programPath: string): Promise<string> {
+    // pkg-dir is an ESM module which we use to find the location of package.json
+    // Because it's an ESM module, we cannot import it directly.
+    const { packageDirectory } = await dynamicImport("pkg-dir");
+    // Check if programPath is a directory. If not, then we
+    // look at it's parent dir for the package root.
+    let isDirectory = false;
+    if(fs.existsSync(programPath)) {
+        const fileStat = await fspromises.lstat(programPath);
+        isDirectory = fileStat.isDirectory();
+    }
+    const programDirectory = isDirectory ? programPath : path.dirname(programPath);
+    const pkgDir = await packageDirectory({
+        cwd: programDirectory,
+    });
+    if (pkgDir === undefined) {
+        log.warn("Could not find a package.json file for the program. Using the pulumi program directory as the project root.");
```

Comment:
nit: s/pulumi/Pulumi
```suggestion
        log.warn("Could not find a package.json file for the program. Using the Pulumi program directory as the project root.");
```